tutorials/3d/using_decals.rst

```diff
@@ -0,0 +1,282 @@
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+1. Create a Decal node in the 3D editor.
+2. In the inspector, expand the **Textures** section and load a texture in
+   **Textures > Albedo**.
+3. Move the Decal node towards an object, then rotate it so the decal is visible
+   (and in the right orientation). If the decal appears mirrored, try to rotate
+   it by 180 degrees. You can double-check whether it's in the right orientation
+   by increasing **Parameters > Normal Fade** to 0.5. This will prevent the Decal
+   from being projected on surfaces that are not facing the decal.
+4. If your decal is meant to affect static objects only, configure it to prevent
+   affecting dynamic objects (or vice versa). To do so, change the decal's
+   **Cull Mask** property to exclude certain layers. After doing this, modify
+   your dynamic objects' MeshInstance3D nodes to change their visibility layers.
+   For instance, you can move them from layer 1 to layer 2, then disable layer 2
+   in the decal's **Cull Mask** property.
+
+Creating decals during gameplay
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Decal nodes can be created from a script like any other node. The main part that can
+be challenging is correctly rotating the decal so it can be visible without
+distortion.
+
+The following code snippet creates a decal at the current mouse cursor location
+when the space bar is pressed:
+
+.. TODO: Add code sample for rotating the decal based on the raycast collision normal.
+
+::
+
+    extends Node3D
+
+    func _input(event):
+        if event.is_action_pressed("ui_accept"):
+            var camera = get_viewport().get_camera_3d()
+            const MAX_DISTANCE = 100
+            var target = camera.project_position(get_viewport().get_mouse_position(), MAX_DISTANCE)
+
+            var query = PhysicsRayQueryParameters3D.create(camera.global_position, target)
+            var result = camera.get_world_3d().direct_space_state.intersect_ray(query)
+
+            if not result.is_empty():
+                var decal = Decal.new()
+                decal.texture_albedo = preload("res://path/to/decal_texture.png")
+
+                # Instead of creating the Decal node in a script, you can instantiate a scene instead.
+                # This allows configuring the Decal and its child nodes (if any) in the editor.
+                #var decal = preload("res://decal.tscn").instantiate()
+
+                add_child(decal)
+
+                decal.position = result["position"]
+                # Rotate the decal to match the camera's orientation.
+                # Depending on your use case, this may make more sense than using the collision normal,
+                # but it can lead to distortion if the camera is nearly parallel to a surface.
+                # To prevent distortion, you'd have to use the collision normal instead to orient the decal
+                # (`result["normal"]`).
+                decal.transform.basis = camera.global_transform.basis
```

Comment:
I think it may be more instructive to have a simpler example. This example is pretty complex and isn't really useful enough to be used in a game as it doesnt cover rotation to align with the normal of the surface. 

I would instead do something that is short and works out of the box. Like leaving footprints behind a player

Reply:
I've simplified the code sample but kept its behavior identical. Hopefully, it should be short enough now.

Reply:
Its not really a matter of length, its a matter of its use as an educational tool. This example is unnecessarily complex for the thing that it is trying to teach. 

The point this section makes is that decals can be dynamically added through code. But the example we give is more like "how to raycast from the mouse position into the scene". The only decal-specific code is:
```
var decal = Decal.new()
decal.texture_albedo = preload("res://path/to/decal_texture.png")
```

I understand that you are trying to insert a real world example in here (which is a great idea), but the example isn't useful in real world projects as it omits rotating based on surface normal.

IMO code examples should either be instructive, or they should be usable in a real project, as-is this code example is neither.

I suggest removing it for the time being, or replacing it with something that is short and self-contained

Reply:
I've removed the section entirely for now. I'll create an issue to track its reimplementation.